cal/cal/models.py

```diff
@@ -717,17 +736,19 @@ def import_calendars(self, only_primary=True):
         self.next_sync_token = result['nextSyncToken']
 
         for item in result['items']:
+            gcal, gcal_created = GCalendar.objects.get_or_create(user=self.user, calendar_id=item['id'])
+
+            # This is the primary calendar, save it as such
             if item.get('primary', False):
-                # This is the primary calendar, save it as such
-                gcal, gcal_created = GCalendar.objects.get_or_create(user=self.user, calendar_id=item['id'])
-                if gcal_created:
-                    gcal.update_meta()
                 profile, _ = Profile.get_or_create(self.user)
                 profile.main_calendar = gcal
                 profile.save()
-            elif not only_primary:
-                gcal, gcal_created = GCalendar.objects.get_or_create(user=self.user, calendar_id=item['id'])
-                if gcal_created:
-                    gcal.update_meta()
+
+            # Update fields for each calendar
+            gcal.summary = item.get('summary')
+            gcal.color_index = item.get('colorId', '1')
```

Comment:
get avoids key errors by returning a default value if a specified key is not found. '1' is the default color, yes, and will be set as such when a colorId key doesn't exist